front/pages/api/w/[wId]/assistant/conversations/[cid]/events.ts

```diff
@@ -0,0 +1,123 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { getConversation } from "@app/lib/api/assistant/conversation";
+import { getConversationEvents } from "@app/lib/api/assistant/pubsub";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { ReturnedAPIErrorType } from "@app/lib/error";
+import { Conversation } from "@app/lib/models";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { ConversationType } from "@app/types/assistant/conversation";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ReturnedAPIErrorType>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you're trying to modify was not found.",
+      },
+    });
+  }
+
+  if (!auth.user()) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_user_not_found",
+        message: "Could not find the user of the current session.",
+      },
+    });
+  }
+
+  if (!auth.isUser()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "workspace_auth_error",
+        message:
+          "Only users of the current workspace can access chat sessions.",
+      },
+    });
+  }
+  if (!(typeof req.query.cId === "string")) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `cId` (string) is required.",
+      },
+    });
+  }
+  const conversationId = req.query.cId;
+  // just checking if the conversation exists, no need to actually construct the
+  // conversation object
+  const conversation = await Conversation.findOne({
```

Comment:
Nope no model in API routes

Should be an Result<> pattern on getConversationEvents

Reply:
Or a new API in lib/api to do that

Reply:
As an example you fail to check the workspace ownership

Let's just get the full conversation that's just simplre